src/Passes/Source/Rules/Factory.cpp

```diff
@@ -375,6 +419,11 @@ namespace quantum
         addRule({call("__quantum__qis__m__body", "qubit"_cap = _), std::move(replace_measurement)});
     }
 
+    void RuleFactory::optimiseBranchQuatumZero()
+    {
+        // TODO(tfr): Not implemented
```

Comment:
Should this throw a not implemented exception for now?

Reply:
I am happy either way. As such it should just be implemented - I just did not find the time for it yet.

Reply:
I would suggest to throw a not-implemented-exception then if the intent is that it should have to have an implementation. 
Also it look like there is another type of Quatum opposed to Quantum here.

Reply:
That's fine. I will add a throw in #1142 for the sake of closing some of these PRs.